internal/api/core/describe.go

```diff
@@ -32,18 +32,24 @@ type APIDescription struct {
 }
 
 type EndpointDescription struct {
-	Description  string             `json:"description"`
-	RequestType  string             `json:"-"`
-	ResponseType string             `json:"-"`
-	Input        []FieldDescription `json:"input"`
-	Output       []FieldDescription `json:"output"`
+	Description  string                 `json:"description"`
+	RequestType  string                 `json:"-"`
+	ResponseType string                 `json:"-"`
+	Input        []FieldDescription     `json:"input"`
+	Output       []BaseFieldDescription `json:"output"`
 }
 
-type FieldDescription struct {
+type BaseFieldDescription struct {
 	Name string `json:"name"`
 	Type string `json:"type"`
 }
 
+type FieldDescription struct {
+	BaseFieldDescription
+
+	Required bool `json:"required"`
```

Comment:
I'm thinking you should use `omitzero` with this.
For required is not a big deal,
but we probably don't want to make a pattern of having all the extra fields for each type.